vllm/v1/worker/gpu_model_runner.py

```diff
@@ -1119,6 +1106,20 @@ def execute_model(
             valid_sampled_token_ids = self.rejection_sampler.parse_output(
                 sampled_token_ids, self.input_batch.vocab_size)
 
+        # TODO(woosuk): The following loop can be slow since it iterates over
+        # the requests one by one. Optimize.
+        for i, generator in self.input_batch.generators.items():
+            req_id = self.input_batch.req_ids[i]
+            req_state = self.requests[req_id]
+            seq_len = (req_state.num_computed_tokens +
+                       scheduler_output.num_scheduled_tokens[req_id])
+            if seq_len < req_state.num_tokens:
+                # Ignore the sampled token for partial prefills.
+                # Rewind the generator state as if the token was not sampled.
+                # This relies on cuda-specific torch-internal impl details
+                generator.set_offset(generator.get_offset() - 4)
+                valid_sampled_token_ids[i] = []
```

Comment:
@comaniac I think you wanted to do
```python
for i, req_id in enumerate(self.input_batch.req_ids):
```

Reply:
Done. Please help double check to make sure the way I got `generator` is correct.